src/News/TwitterWatcher.php

```diff
@@ -81,23 +83,26 @@ protected function notifyTwitter(Message $event, Message $article, Pbjx $pbjx):
             return;
         }
 
-        $app = $this->getApp($article, $event, $pbjx);
-        if (null === $app) {
+        $apps = $this->getApps($article, $event, $pbjx);
+        if (null === $apps || count($apps) === 0) {
             return;
         }
 
-        $notification = $this->createTwitterNotification($event, $article, $pbjx)
-            ->set('app_ref', $app->generateNodeRef());
-
-        $command = CreateNodeV1::create()->set('node', $notification);
-        $pbjx->copyContext($event, $command);
-        $nodeRef = $article->generateNodeRef();
-
-        try {
-            $pbjx->sendAt($command, strtotime('+3 seconds'), "{$nodeRef}.post-tweet");
-        } catch (\Throwable $e) {
-            if ($e->getCode() !== Code::ALREADY_EXISTS) {
-                throw $e;
+        foreach ($apps as $app) {
+            $appRef = NodeRef::fromNode($app);
+            try {
+                $notification = $this->createTwitterNotification($event, $article, $app, $pbjx)
+                    ->set('app_ref', $appRef);
+
+                $command = CreateNodeV1::create()->set('node', $notification);
+                $pbjx->copyContext($event, $command);
+                $nodeRef = $article->generateNodeRef();
+
+                $pbjx->sendAt($command, strtotime('+3 seconds'), "{$nodeRef}.{$appRef}.post-tweet");
```

Comment:
@meghedi @efjacobson only concern here with the mega job id is string length.  maybe use `$pbjx->sendAt($command, strtotime('+3 seconds'), "{$nodeRef}.{$appRef->getId()}.post-tweet");`

Reply:
Done.